go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1121,7 +1121,11 @@ func (npw *nodePortWatcher) AddEndpointSlice(epSlice *discovery.EndpointSlice) e
 	out, exists := npw.getAndSetServiceInfo(*svcNamespacedName, svc, hasLocalHostNetworkEp, localEndpoints)
 	if !exists {
 		klog.V(5).Infof("Endpointslice %s ADD event in namespace %s is creating rules", epSlice.Name, epSlice.Namespace)
-		return addServiceRules(svc, netInfo, sets.List(localEndpoints), hasLocalHostNetworkEp, npw)
+		if err = addServiceRules(svc, netInfo, sets.List(localEndpoints), hasLocalHostNetworkEp, npw); err != nil {
```

Comment:
in update path are we covered?

Reply:
Good point, previously it wasn't handled either.
I fixed it by using a `getServiceInfo` with `addOrSetServiceInfo`/`updateServiceInfo` instead of the `getAndSetServiceInfo`